replica/tablets.cc

```diff
@@ -252,6 +252,10 @@ future<tablet_metadata> read_tablet_metadata(cql3::query_processor& qp) {
             for (auto&& r : tablet_replicas) {
                 pending.erase(r);
             }
+            if (pending.empty() && transition != tablet_transition_kind::rf_change) {
```

Comment:
Why is that? The code below is well prepared for empty pending replica for any transition kind

Reply:
This change was based on this comment https://github.com/scylladb/scylladb/pull/16723#issuecomment-1979357804
IIUC I can remove this `if` now, right?

Reply:
OK, removed.